packages/docs/docs/providers/saml-jackson.md

```diff
@@ -0,0 +1,58 @@
+---
+id: saml-jackson
+title: BoxyHQ SAML Jackson
+---
+
+## Documentation
+
+BoxyHQ SAML Jackson is an open source service that handles the SAML login flow as an OAuth 2.0 flow, abstracting away all the complexities of the SAML protocol.
+
+You can deploy SAML Jackson as a separate service or embed it into your app using our NPM library. [Please check our documentation for more details](https://boxyhq.com/docs/jackson/deploy)
+
+## Configuration
+
+SAML login requires a configuration for every tenant of yours. One common method is to use the domain for an email address to figure out which tenant they belong to. You can also use a unique tenant ID (string) from your backend for this, typically some kind of account or organization ID.
+
+Check our [documentation](https://boxyhq.com/docs/jackson/getting-started#2-saml-config-api) for more details.
+
+## Options
+
+The **SAML Jackson Provider** comes with a set of default options:
+
+- [BoxyHQ Provider options](https://github.com/nextauthjs/next-auth/blob/main/src/providers/boxyhq.ts)
+
+You can override any of the options to suit your own use case.
+
+## Example
+
+```ts
+import SAMLJacksonProvider from "next-auth/providers/saml-jackson"
+...
+providers: [
+  SAMLJacksonProvider({
+    issuer: "http://localhost:5000",
+    clientId: "dummy", // The dummy here is necessary since we'll pass tenant and product custom attributes in the client code
+    clientSecret: "dummy", // The dummy here is necessary since we'll pass tenant and product custom attributes in the client code
+  })
```

Comment:
```suggestion
  SAMLJacksonProvider({ issuer: process.env.SAML_JACKSON_ISSUER })
```

This is an unusual flow, so I would like to have more context here. Why are client id and client secret "dummy" values?

If the `clientId` and `clientSecret` values aren't important somehow (very curious), we should just omit them, or set them by default in the provider code.

From the localhost issuer, I will also assume that you don't have a hosted service/it's something the user will always need to set up themselves? Do you have something similar to https://demo.identityserver.io/ that the user could play with? It would be a nice mention on this docs page.

Is this referring to the same as here: https://github.com/nextauthjs/next-auth/pull/3782/files#diff-e4ec1563a89a51e2b61e12a89fc66bb226e55e3f27173919bdb9b9f75a713d85R10

Reply:
`clientId` and `clientSecret` aren't as critical here (unlike OAuth 2.0) because SAML is tenanted (unique config per customer, that is also configured by the customer using their SAML Identity Provider). When `tenant` and `product` are used then `clientId` and `clientSecret` become merely artifacts for the OAuth 2.0 flow. More details explained [here](https://boxyhq.com/docs/jackson/security#client-secret).

We just released a mock SAML service that can be used for testing - https://mocksaml.com, I'll add it to the docs. Please also try our demo here to get an understanding of the flow - https://saml-demo.boxyhq.com/

Reply:
I had originally internalised the `dummy` logic here - https://github.com/nextauthjs/next-auth/pull/3782/commits/8955ffd887a63fc40b670ed4251488e2c608e177#diff-76f06144da0418fa936f7f4af37fcf4b1c21448f94716360feef15e4c2be4013L46

Reply:
@balazsorban44 I have setup a demo that requires no configuration from your side. Please run the app from this branch - https://github.com/nextauthjs/next-auth/pull/3782/commits/afbdc5b4aee33ccc4ef1f5292ca44217a5fe94c3